File: py_solutions_21-30/Euler_22.py
# Number amount of names

from __future__ import print_function

from operator import add
from functools import reduce

import timeit
import os


start = timeit.default_timer()

def euler_22():
    with open(os.path.abspath('').strip('py_solutions_21-30') + 
                                        '/euler_txt/names1.txt') as f:
        name_list = f.read()
    name_list = name_list.replace('"', '')
    name_list = sorted(name_list.split(','))
    return reduce(add, [v * sum(map(lambda x: ord(x) - 64, k))
                        for v, k in enumerate(name_list, 1)])

print("Answer: {}".format(euler_22()))
stop = timeit.default_timer()
print("Time: {0:9.5f}".format(stop - start))

File: perl_solutions_21-30/Euler_22.pl
use strict;
use warnings;
use Time::HiRes qw( clock );


sub name_sum
{
    my $total = 0;
    my $pos = 1;
    open(my $f, "/home/tijko/Project_Euler/euler_txt/names1.txt");
    my @name_list = sort((split('"', <$f>)));
    for my $name (@name_list) {
        if (length($name) > 1) {
            my $name_total = 0;
            for my $letter (split(//, $name)) {
                $name_total += (ord($letter) - 64);
            }
            $total += ($name_total * $pos);
            $pos++;
        }
    }
    return $total;
}

my $start = clock();
my $answer = name_sum();
my $stop = clock();
print "Answer: $answer\n";
printf "Time: %.5f\n", $stop - $start;
